### PR TITLE
Draft: Add Tags

### PR DESCRIPTION
**Pull request by flacnut**
*I am not familiar with Ruby or Rails, constructive feedback welcomed!*

This PR is a draft of possible ways to add tags.

[Context (713)](https://github.com/maybe-finance/maybe/discussions/713)

I have in the past used the ORM / ActiveRecord completely, mapping a many-to-many with a join table, however this negatively impacted the performance of my application for the use case of generating many charts which include tag filters.

Tags_demo_data_2

### CODE REVIEW

File: app/controllers/accounts_controller.rb [hunk 1]
```diff
@@ -82,7 +82,11 @@ def sync
   private
 
   def set_account
-    @account = Current.family.accounts.find(params[:id])
+    begin
+      @account = Current.family.accounts.find(params[:id])
```
Comment (zachgoll) [inline app/controllers/accounts_controller.rb:86]: In Rails this error handling is typically not necessary as the `@account` would just be `nil` if not found.

**Review by zachgoll — COMMENTED** [review]
Appreciate the PR here, as I mentioned in the [Discussion](https://github.com/maybe-finance/maybe/discussions/713#discussioncomment-9330362), I definitely think tags are a must-have for the system.

We don't have any internal designs for this quite yet, so I think in order to introduce this, it would have to be _backend only_ for the moment.

If you wanted to go down that route, I'd probably favor a many to many relationship as ActiveRecord has a [lot of neat helpers](https://edgeguides.rubyonrails.org/association_basics.html#methods-added-by-has-and-belongs-to-many) and handles this quite well.

File: app/controllers/accounts_controller.rb (hunk 1 above)
Comment (flacnut) [reply app/controllers/accounts_controller.rb:86]: Yeah, this was actually a fix for something else. If you navigate to an `account_id` that doesn't exist, rails spits out stack traces which is not very graceful.

**Comment by zachgoll** [comment]
@flacnut going to close this out for now and we can re-open it (or open a new one) when we get the designs for it.  We've been doing some roadmap planning and will definitely be getting this in soon!

**Pull request closed without merging**